sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -403,13 +404,11 @@ object CatalogTypes {
 /**
  * A [[LogicalPlan]] that represents a table.
  */
-case class CatalogRelation(
+case class CatalogRelation private(
     tableMeta: CatalogTable,
     dataCols: Seq[AttributeReference],
     partitionCols: Seq[AttributeReference]) extends LeafNode with MultiInstanceRelation {
   assert(tableMeta.identifier.database.isDefined)
-  assert(tableMeta.partitionSchema.sameType(partitionCols.toStructType))
-  assert(tableMeta.dataSchema.sameType(dataCols.toStructType))
```

Comment:
remove these 2 asserts as this can be guaranteed by `CatalogRelation.apply(tableMeta: CatalogTable)`